### DISCUSSION TITLE
How do I show "small" column value for 0 entry in series?

### DISCUSSION THREAD

**Discussion by houstonhaynes** [Live-Charts/LiveCharts2#1089] [Q&A]
I'd like for a column series to show a *slight* value (less than 1) when a 0 value is present in the series (as opposed to `Nullable()` which I expect to be fully blanked). Is there a setting that tells the chart to render a tiny bit of column height with a 0 value?

![Screenshot 2023-06-30 at 9 57 29 AM](https://github.com/beto-rodriguez/LiveCharts2/assets/8174976/cfe23436-6e80-47be-a112-fd791d344d35)

**Comment by houstonhaynes** [comment]
As an experiment I tried ChatGPT and it gave some interesting info without actually resolving the problem. It seems to have done what it's known for - mixing up APIs and getting some of the details wrong. What was useful to me was the explanation that came with it. I'm working in F# so that's what I'll quote here. 

But the curious thing that's pointed out is below that there needs to be a "peeling back" of the API a bit to reconstruct a custom series config that applies 

```fsharp
let updateSeriesFill (series: ColumnSeries<DateTimePoint>) =
    let zeroLinePaint = new SKPaint()
    zeroLinePaint.Style <- SKPaintStyle.Stroke
    zeroLinePaint.Color <- SKColors.Red
    zeroLinePaint.StrokeWidth <- 1.0f
    
    series.CustomData <-
        series.Values
        |> Seq.map (fun value ->
            match value.Value with
            | Some y when y = 0.0 ->
                value, (fun (ctx: DrawingContext) ->
                    ctx.Canvas.DrawLine(
                        value.CanvasPosition,
                        value.CanvasPosition + SKPoint(0.0f, -0.1f),
                        zeroLinePaint
                    )
                )
            | _ -> value, (fun _ -> ()))

    let chart = series.ParentChart
    chart.Series <- chart.Series.Add(series)

```

which also give errors for not resolving 'CustomData' or 'ParentChart'

> I apologize for the confusion. It appears that the specific property CustomData does not exist in the ColumnSeries type in LiveChartsCore.
> 
> To clarify, LiveChartsCore version 2.0.0-beta.802 does not provide a built-in property or mechanism called CustomData in the ColumnSeries type. If you require custom data storage or additional properties for each data point in a column series, you may need to implement your own custom solution or extend the library's functionality.
> 
> If you have any further questions or need assistance with any other topic, please let me know.

**Reply by houstonhaynes** [reply]
OK so I'm "cheating" - setting all 0 values to "0.25" and then flattening the tooltip to show the `chartPoint.SecondaryValue` as and `int`value

![Screenshot 2023-06-30 at 6 34 04 PM](https://github.com/beto-rodriguez/LiveCharts2/assets/8174976/55d7983b-6b45-4423-b365-f63b86606104)

Well at least I'm doing that in code... 

```fsharp
let tooltipFormatter (chartPoint: ChartPoint) =
    $"{chartPoint.Context.Series.Name}: {chartPoint.SecondaryValue}, {int chartPoint.PrimaryValue}"

let rec init() =
    {
        Series = 
            ObservableCollection<ISeries> 
                [ 
                    let columnSeries = ColumnSeries<DateTimePoint>(
                        Values = newSeries(None),
                        Name = "Luck By Second",
                        TooltipPointFormatter = tooltipFormatter
                    )
                    columnSeries :> ISeries
                ]
        Actions = [ { Description = "Initialized Chart"; Timestamp = DateTime.Now } ]
        IsAutoUpdateChecked = false
    }
```

but I have yet to see it show up in the chart. I have a feeling I'll need to define a "CustomToolTip" and then put a binding out to the AXAML, which I'll save for another day. 

If anyone has any *OTHER* ideas I'd be happy to read them. My preference is to *not* manipulate the series in order to accomplish this. I'd rather the presentation layer handle it *for me* without massaging the data. Ideas welcome! Cheers!

**Reply by beto-rodriguez** [reply]
Interesting case, I will keep this in mind and come back if find/think of a solution for this.

**Reply by houstonhaynes** [reply]
Thanks. There are some options for this in D3/Plotly and Highcharts - both of which I've built in the R ecosystem. The idea of "just paint a line" gets wrinkly when there are formatting options applied to the "main" series. That can turn into a mess once you start down the path of chasing two separate render methods.

The same could be said for generating a second series to paint the background for 'nullables' (so that you can see 0s because they're *not* shaded like the nulls). It turns into a slippery slope of "nibbling around the edges" until it's its own engineering project.

And of course, the final irony is the line series can put a marker right on the axis. So that's "already done". 😏
